lib/src/kbasesearchengine/events/storage/MongoDBStatusEventStorage.java

```diff
@@ -227,8 +227,8 @@ private StoredStatusEvent toEvent(final Document event) {
     public List<StoredStatusEvent> get(final StatusEventProcessingState state, int limit)
             throws FatalRetriableIndexingException {
         Utils.nonNull(state, "state");
-        if (limit < 1 || limit > 1000) {
-            limit = 1000;
+        if (limit < 1 || limit > 10000) {
```

Comment:
Hmm, I'm not sure I like the idea of putting implementation code (e.g. a constant) in an interface. My feeling is that the implementation should determine how it wants to store the limit. In this case since the limit is localized to two concurrent lines in the same function I didn't bother making an implementation constant, but I'm happy to do so. What do you think?

Reply:
Fine with me, but how about putting the constant in MongoDBStatusEventStorage.java. I think you have 10k defined in SearchTools as well and it looks related. If it is, it would be better to just reference the 10k constant from MongoDBStatusEventStorage.java

Reply:
I made the maximum queue size the same size as the maximum number of records that can be pulled from storage at once as a convenience, but there's no real relation.

If I promise to make the change in a future PR can we merge this? locally I'm several PRs ahead so I'd have to branch and cherry pick to fix in this PR.